### DISCUSSION TITLE
Regarding the issue of UWB positioning base station building occlusion.

### DISCUSSION THREAD

**Discussion by lalalalang999** [bitcraze/discussions#483] [Q&A]
hi:
I have a question about setting up a base station. Using the TDOA2 algorithm, when building a UWB base station, it is necessary to satisfy that the 0 base station (main base station) is visible to all base stations. According to the principle of the TDOA algorithm, it is also necessary to ensure that adjacent base stations are visible, such as base station 0 and base station 1, base station 1 and base station 2, base station 2 and base station 3... I have a question, when building a base station, is it not It is necessary to ensure that non-adjacent base stations are visible. For example, there is occlusion between base station 1 and base station 3, which cannot be seen.

In other words, assuming that I need to build 6 base stations, I can ensure that base station 0 and all base stations are visible, and every two adjacent base stations are visible. But I can't guarantee that all base stations are visible to each other. For example, base station No. 1 cannot see base station No. 4. Will this affect the accuracy of the TDOA2 algorithm?

Best Regards

**Comment by krichardsson** [comment]
Hi,
If I remember correctly the anchors are used in pairs and must be visible to each other (I'm not completely sure). This means anchor 0 and anchor 1 must see each other, 1 and 2, 2 and 3, 3 and 4, ... 7 and 0.
As you mention 0 must be visible to all anchors since this is used for time synchronization of the protocol.

I would strongly suggest to switch to TDoA3 if possible. If all anchors can not see each other TDoA3 will make life much easier and it is also possible that the position estimate will be better as the system can utilize the available data in a better way. Also if your space is large, you can switch to Long range which will increase the range.

**Reply by lalalalang999** [reply]
hi,
That means that in the TDOA2 algorithm, if base station 1, base station 3, and base station 4 are invisible, it will have no effect.

**Reply by krichardsson** [reply]
The Crazyflie will use the information that is available and if it does not get any information about some anchors, it will use the others.

**Reply by lalalalang999** [reply]
hi:
Maybe there's something wrong with my statement. What I'm confused about is that when constructing ANCHOR with TDOA2, if ANCHOR1 and ANCHOR3, or ANCHOR1 and ANCHOR4, are invisible. Is not visible between ANCHOR, not between indicator tag and ANCHOR. I just want to ensure that ANCHOR0 is visible to all anchors and adjacent anchors. Is it feasible to set up the ANCHOR system in this way?

**Reply by knmcguire** [reply]
hi! Kristoffer is still away for a bit, but I'm afraid that your explanation is not clear to us. At least for TDOA2, if the adjacent pairs are not seen by eachother, than it will not work at all as it works in a round robin fashion.

Anyway, we have a lot of documentation on the matter which you can find here: https://www.bitcraze.io/documentation/repository/lps-node-firmware/master/. If that doesn't asnwer your question, please come back here or make a new discussion with a more formulated question (perhaps with a drawing or such).
